### PR TITLE
Feature: indeed connector to pull job offers into HrFlow boards..

### PR DESCRIPTION
**Pull request by limamvadhel**
This merge request will add in the hrflow_connectors boards a new connector for Indeed job postings:

- **actions.py** which contains the scripts to run to get a custom choice of  job offers from Indeed.
- The file contains one class, the **"GetAllJobs"** class, that inherits **"BoardAction"**classes , which is divided into a  **"Crawler"** function to run the chromium driver which is mandatory to **scrap** and **parse** jobs with **selenium**, a **"pull"** function that run through all the indeed pages customized by the user and scrap each job card and each job link and a **"format"** function that formats parsed attributes of job offers into the HrFlow job object format.

-It will also add inside the tests folder, a file **test_get_all_jobs_indeed.py**  to test the connector.

### CODE REVIEW

File: src/hrflow_connectors/connectors/boards/indeed/actions.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+                # get joblink for each jobCard
+                elements = jobCards.find_elements_by_class_name("tapItem")
+                for x in elements:
+
+                    link = x.get_attribute("href")
+                    jobs_Links.append(link)
+
+            finally:
+
+                pass
```
Comment (CorentinDuchene) [inline src/hrflow_connectors/connectors/boards/indeed/actions.py:120]: * `except` a specific Exception, not a generic Exception. So you have to remove the `finally`
* `jobs_Links` -> `jobs_links` or `jobs_links_list` (snake_case)

**Review by CorentinDuchene — COMMENTED** [review]
First review of the Crawler, the connector (everything but `format`)

File: src/hrflow_connectors/connectors/boards/indeed/actions.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        # Get total number of related job offers in all pages.
+        total_job_s = driver.find_element_by_id("searchCountPages").text
+
+        # retrieve the number of total related job offers from string 'for example from Page 1 de 993 emplois we get total_jobs = 993'
+        total_job_s = total_job_s.split()
+        start = total_job_s.index("de")
+        end = total_job_s.index("emplois")
+        total_jobs = int("".join([total_job_s[i] for i in range(start + 1, end)]))
+
+        count_jobs = self.limit  # max 15 jobs per page
```
[the remark this answers is not in the document]
Comment (limamvadhel) [reply src/hrflow_connectors/connectors/boards/indeed/actions.py:97]: I have tried to play with page limit and make it optional but the script doesn't like that, so i got rid of it temporarily until we get a better option.

File: src/hrflow_connectors/connectors/boards/indeed/actions.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        total_page = int(total_jobs / count_jobs)
+
+        for page in range(0, total_page):
+            if page == self.limit_extract:  # break if page reaches limit set by user
```
[the remark this answers is not in the document]
Comment (limamvadhel) [reply src/hrflow_connectors/connectors/boards/indeed/actions.py:101]: I have tried to play with page limit and make it optional but the script doesn't like that, so i got rid of it temporarily until we get a better option.

**Review by CorentinDuchene — APPROVED** [review]

**Pull request merged**